extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -121,23 +124,22 @@ export class AmpAutocomplete extends AMP.BaseElement {
     this.maxEntries_ = this.element.hasAttribute('max-entries') ?
       parseInt(this.element.getAttribute('max-entries'), 10) : null;
 
-    return this.mutateElement(() => {
-      this.container_ = this.createContainer_();
-      this.element.appendChild(this.container_);
-    });
+    this.container_ = this.createContainer_();
+    this.element.appendChild(this.container_);
+
+    return dataPromise.then(data => this.inlineData_ = data);
```

Comment:
Sorry to comment on this again, but we should really not make any XHR call from the buildCallback.
`buildCallback` can be called when the page is being prerendered, but not visible to the user. Pinging the publisher's server before the user clicks on a link to view the page would break privacy rules.
Please move this call to the `layoutCallback`.

Reply:
That's right, sorry I missed this comment!

Reply:
PTAL. I don't like that both callbacks check for the `src` attribute, but avoiding that means moving the call to `getInlineData_` to the `layoutCallback` also, which seems unnecessary to me.